components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java

```diff
@@ -0,0 +1,5 @@
+package org.apacheextras.camel.component.wmq;
+
+public class MQHeaderListCreationException extends RuntimeException {
```

Comment:
Why not using a `IllegalStateException` instead of a custom one ?

Reply:
When I started writing Junit test, I found a very complex error linked to MQHeaderList (when I create a new MQHeaderList instance). 

I think that this specific exception can help to understand the problem (if you forget a class in the classpath, currently not in the module dependency, you get a NoSucheMethodError).

I can add more information about that in the exception message to help developers to understand that problem.

Reply:
And i forgot the super call so I need to update it

Reply:
You are right, I wanted to handle checked exception and IllegalStateException can be enough.

I add another exception message for a case really hard to debug that I found when I wrote my unit cases (and it's the reason why I added the HeaderListFactory)

Reply:
+1

Reply:
can you wait before merging ? I want to do one more test before it

Reply:
Sure, no problem, I won't merge before the commits squashing.